rerun-io/rerun
# Introduce kittest for automated UI testing and git-lfs for UI snapshot handling

### What

- Integrates egui_kittest for automated UI testing, including snapshot testing
- Update egui to latest main
- Introduces git-lfs to the repo.
  - Add instructions in `BUILD.md`
  - Improve `check_large_files.py` to verify that all PNGs are indeed handled by `git-lfs`.
- Run mac rust tests on `main` to cover for the UI snapshot tests
- Related: https://github.com/rerun-io/rerun/issues/8245

**NOTE**: I (@abey79) repeat: introduce git-lfs to the repo.

### Checklist
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/8182?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/8182?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide
* [x] Run tests in CI
* [x] Setup git lfs

- [PR Build Summary](https://build.rerun.io/pr/8182)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

To deploy documentation changes immediately after merging this PR, add the `deploy docs` label.
